.github/workflows/build.yml

```diff
@@ -0,0 +1,39 @@
+name: Java CI with Gradle
+
+on: [push, pull_request]
+
+jobs:
+  build:
+    # Only run on PRs if the source branch is on someone else's repo
+    if: "${{ github.event_name != 'pull_request' || github.repository != github.event.pull_request.head.repo.full_name }}"
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository
+        # See https://github.com/actions/checkout/commits
+        uses: actions/checkout@3df4ab11eba7bda6032a0b82a6bb43b11571feac
+
+      - name: Validate Gradle wrapper
+        # See https://github.com/gradle/wrapper-validation-action/commits
+        uses: gradle/wrapper-validation-action@342dbebe7272035434f9baccc29a816ec6dd2c7b
+
+      - name: Cache Gradle packages
+        uses: actions/cache@v3
+        with:
+          path: |
+            ~/.gradle/caches
+            ~/.gradle/wrapper
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*', '**/gradle-wrapper.properties') }}
+          restore-keys: |
+            ${{ runner.os }}-gradle-
+
+      - name: Set up JDK 17
+        # See https://github.com/actions/setup-java/commits
+        uses: actions/setup-java@0ab4596768b603586c0de567f2430c30f5b0d2b0
+        with:
+          java-version: 17
+          distribution: temurin
+
+      - name: Build with Gradle
```

Comment:
Please use the gradle build action and validate the gradle build wrapper. See Geyser for an example.

Reply:
But I do validate the gradle build wrapper there...
I'll move the action.